components/brave_wallet/browser/eth_abi_decoder.h

```diff
@@ -0,0 +1,26 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_ETH_ABI_DECODER_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_ETH_ABI_DECODER_H_
+
+#include <string>
+#include <vector>
+#include "base/values.h"
+
+namespace brave_wallet {
+
+bool UniswapEncodedPathDecode(const std::string& encoded_path,
+                              // NOLINTNEXTLINE(runtime/references)
+                              std::vector<std::string>& path);
+
+bool ABIDecode(const std::vector<std::string>& types,
+               const std::string& data,
+               std::vector<std::string>* tx_params,
```

Comment:
for multiple outputs use `absl::optional<std::tuple>`

Reply:
Fixed here: a22490327c9e3c5e786bf8c9d888ccb729df2571